src/pybind/mgr/dashboard/frontend/cd.js

```diff
@@ -46,6 +46,7 @@ function prepareLocales() {
   }
 
   let langs = process.env.DASHBOARD_FRONTEND_LANGS || '';
+  langs = langs.split('"').join('')
```

Comment:
Can you please put an example of `langs` that makes this necessary? I assume that the env is `cs,de,es,fr,id,it,ja,ko,pl,zh-Hans,zh-Hant,pt`, isn't it?

Reply:
> Can you please put an example of `langs` that makes this necessary? I assume that the env is `cs,de,es,fr,id,it,ja,ko,pl,zh-Hans,zh-Hant,pt`, isn't it?

The env is `cs,de,es,fr,id,it,ja,ko,pl,zh-Hans,zh-Hant,pt` or `"ALL"`, When we need to supports multiple languages, it needs to be `"ALL"` without specifying each language. And More Information is in https://github.com/ceph/ceph/blob/main/src/pybind/mgr/dashboard/frontend/cd.js#L49.

Reply:
What does it have to do with removing quotes? If that's the goal, `.replace()` should be a better alternative. But I still don't get it, `DASHBOARD_FRONTEND_LANGS="ALL"` in the shell will be just `ALL` in Javascript. Isn't that the case? This would only happen if `DASHBOARD_FRONTEND_LANGS='"ALL"'` or `DASHBOARD_FRONTEND_LANGS="\"ALL\""`

Reply:
There use the command about `DASHBOARD_FRONTEND_LANGS="${DASHBOARD_FRONTEND_LANGS}" npm run build:localize` (Please see https://github.com/ceph/ceph/blob/main/src/pybind/mgr/dashboard/frontend/CMakeLists.txt#L119) to build, and The variable `DASHBOARD_FRONTEND_LANGS` is ALL, So The complete command is `DASHBOARD_FRONTEND_LANGS="ALL" npm run build:localize`, I print the value of `process.env.DASHBOARD_FRONTEND_LANGS` in cd.js and get `"ALL"`,I just remove the quotes marks to increase the compatibility, regardless of whether the value about DASHBOARD_FRONTEND_LANGS is ALL or "ALL", Using `.split('"').join('')` to remove quotes is my common way. To make the code clearer, I will replace it with `.replace`.

Reply:
Weird. Maybe Cmake handles quotes differently from the shell. Ok, let's sanitize the string. And thanks for adopting replace. That way you can both replace single and double quotes.